packages/components/src/utils/strings.ts

```diff
@@ -6,64 +6,22 @@ import { paramCase } from 'change-case';
 
 const ALL_UNICODE_DASH_CHARACTERS = new RegExp(
 	`[${ [
-		// - (hyphen-minus)
-		'\u002d',
 		// ~ (tilde)
 		'\u007e',
 		// ­ (soft hyphen)
 		'\u00ad',
-		// ֊ (armenian hyphen)
-		'\u058a',
-		// ־ (hebrew punctuation maqaf)
-		'\u05be',
-		// ᐀ (canadian syllabics hyphen)
-		'\u1400',
-		// ᠆ (mongolian todo soft hyphen)
-		'\u1806',
-		// ‐ (hyphen)
-		'\u2010',
-		// non-breaking hyphen)
-		'\u2011',
-		// ‒ (figure dash)
-		'\u2012',
-		// – (en dash)
-		'\u2013',
-		// — (em dash)
-		'\u2014',
-		// ― (horizontal bar)
-		'\u2015',
 		// ⁓ (swung dash)
 		'\u2053',
-		// superscript minus)
+		// ⁻ (superscript minus)
 		'\u207b',
-		// subscript minus)
+		// ₋ (subscript minus)
 		'\u208b',
 		// − (minus sign)
 		'\u2212',
-		// ⸗ (double oblique hyphen)
-		'\u2e17',
-		// ⸺ (two-em dash)
-		'\u2e3a',
-		// ⸻ (three-em dash)
-		'\u2e3b',
-		// 〜 (wave dash)
-		'\u301c',
-		// 〰 (wavy dash)
-		'\u3030',
-		// ゠ (katakana-hiragana double hyphen)
-		'\u30a0',
-		// ︱ (presentation form for vertical em dash)
-		'\ufe31',
-		// ︲ (presentation form for vertical en dash)
-		'\ufe32',
-		// ﹘ (small em dash)
-		'\ufe58',
-		// ﹣ (small hyphen-minus)
-		'\ufe63',
-		// － (fullwidth hyphen-minus)
-		'\uff0d',
+		// any other Unicode dash character
+		'\\p{Pd}',
 	].join( '' ) }]`,
-	'g'
+	'gu'
 );
 
 export const normalizeTextString = ( value: string ): string => {
```

Comment:
FWIW this function is used only in `ComboboxControl`. Should we just inline it there?